eris_test.go

```diff
@@ -76,61 +109,63 @@ func TestErrorWrapping(t *testing.T) {
-	err error
-}
-
-func (e *withEmptyLayer) Error() string { return e.err.Error() }
-func (e *withEmptyLayer) Unwrap() error { return e.err }
-
 func TestExternalErrorWrapping(t *testing.T) {
 	tests := map[string]struct {
 		cause  error    // root error
 		input  []string // input for error wrapping
-		output error    // expected output
+		output []string // expected output
 	}{
 		"no error wrapping with a third-party root cause (errors.New)": {
-			cause:  errors.New("external error"),
-			output: eris.New("external error"),
+			cause: errors.New("external error"),
+			output: []string{
+				"external error",
+			},
 		},
 		"standard error wrapping with a third-party root cause (errors.New)": {
-			cause:  errors.New("external error"),
-			input:  []string{"additional context", "even more context"},
-			output: eris.Wrap(eris.Wrap(eris.New("external error"), "additional context"), "even more context"),
+			cause: errors.New("external error"),
+			input: []string{"additional context", "even more context"},
+			output: []string{
+				"even more context: additional context: external error",
+				"additional context: external error",
+				"external error",
+			},
 		},
 		"wrapping a wrapped third-party root cause (errors.New and fmt.Errorf)": {
-			cause:  fmt.Errorf("additional context: %w", errors.New("external error")),
-			input:  []string{"even more context"},
-			output: eris.Wrap(eris.Wrap(eris.New("external error"), "additional context"), "even more context"),
+			cause: fmt.Errorf("additional context: %w", errors.New("external error")),
+			input: []string{"even more context"},
+			output: []string{
+				"even more context: additional context: external error",
+				"additional context: external error",
+				"external error",
+			},
 		},
 		"wrapping a wrapped third-party root cause (multiple layers)": {
-			cause:  fmt.Errorf("even more context: %w", fmt.Errorf("additional context: %w", errors.New("external error"))),
-			input:  []string{"way too much context"},
-			output: eris.Wrap(eris.Wrap(eris.Wrap(eris.New("external error"), "additional context"), "even more context"), "way too much context"),
+			cause: fmt.Errorf("even more context: %w", fmt.Errorf("additional context: %w", errors.New("external error"))),
+			input: []string{"way too much context"},
+			output: []string{
+				"way too much context: even more context: additional context: external error",
+				"even more context: additional context: external error",
+				"additional context: external error",
+				"external error",
+			},
 		},
 		"wrapping a wrapped third-party root cause that contains an empty layer": {
-			cause:  fmt.Errorf(": %w", errors.New("external error")),
-			input:  []string{"even more context"},
-			output: eris.Wrap(eris.Wrap(eris.New("external error"), ""), "even more context"),
+			cause: fmt.Errorf(": %w", errors.New("external error")),
+			input: []string{"even more context"},
+			output: []string{
+				"even more context: : external error",
```

Comment:
this looks messy, do we have to retain an empty wrap message in format?

Reply:
i agree it looks messy but it matches the behavior of Golang errors and pkg/errors. i'm open to discussing it but i'm leaning toward allowing users to do things like this and i want print errors as accurately as possible (i.e. i don't like the idea of skipping over empty messages because then what would we do for the `withTrace` case? skip over the stack frame as well?). we can discuss more on discord if you disagree though

Reply:
I am fine with it, another way could be to empty message via preselected tag like <empty_wrap_msg>

Reply:
possibly, just have to keep in mind that we're talking about rare corner cases here that should never happen under normal use